python/cudf/cudf/_libxx/copying.pyx

```diff
@@ -34,3 +39,33 @@ def gather(Table source_table, Column gather_map):
         column_names=source_table._column_names,
         index_names=source_table._index._column_names
     )
+
+
+def shift(Table input, int offset):
+    cdef table_view c_input = input.data_view()
```

Comment:
I believe we'd want to shift the index as well where we'd probably want to use `.view()` and then add `input._index_names` below.

Reply:
Should we force index fill values to be invalid/null?

Reply:
I was actually completely incorrect here: We don't want to shift the index values: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.shift.html

Reply:
But we _do_ want to include them in the results, yes?

Reply:
Yes, we'd want to maintain the index / index names in the result.

Reply:
Going to make some changes to the libcudf++ api so we can support this easier on the python side (and it'll be beneficial to libcudf). Should be pretty simple changes, as we're processing each column individually.

Reply:
#4299

Reply:
Not getting much love on the C++ PR, so I may revert to the table-based shift api and account for the index non-shift on the python side.